docs/state-sync.md

```diff
@@ -0,0 +1,134 @@
+# State Sync
+
+## What is State sync?
```

Comment:
Sorry. One last thing. This heading should be removed. The area under `# State Sync` is there to answer that question anyway. So the next paragraph is just fine directly under it. And then `make update-tocs` after removing it.